tests/test_generate.py

```diff
@@ -0,0 +1,85 @@
+import subprocess
+import pytest
+import os
+from datetime import datetime
+import json
+today = datetime.now().strftime('%Y-%m-%d')
+
+cell_front_matter ={
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "<!-- ---\n",
+                "title: <required-title>\n",
+                f"date: {today}\n",
+                "downloads: true\n",
+                "weight: <required-weight> See: https://github.com/pytorch-ignite/examples/issues/30\n",
+                "summary: <use either this or the `<!--more-->` tag below to provide summary for this notebook, "
+                "and delete the other>\n"
+                "tags:\n",
+                "  - <required-tag>\n",
+                "--- -->\n",
+                "\n",
+                "# title-placeholder\n",
+                "\n",
+                "<If you are not using the `summary` variable above, use this space to "
+                "provide a summary for this notebook.>\n",
+                "<Otherwise, delete the `<!--more-->` below.>",
+                "\n",
+                "<!--more-->",
+            ],
+        }
+ 
+
+cwd = os.getcwd()
+
+
+@pytest.mark.parametrize("name", ["test_notebooks/dummy_notebook", "test_notebooks/dummy_notebook.ipynb"])
+def test_new_notebook_creation(name):
```

Comment:
If we execute this test locally, won't it remove the file from git history ?

Reply:
Yeah, it first creates `dummy_notebook.ipynb` then deletes it. So no change in git history right?

On the other hand, the other tests ``test_existing_non_empty_notebook`` and ``test_existing_blank_notebook`` will add frontmatter to the dummy notebooks, so that needs to be undone else it will show up in git history.

A solution I was thinking of was, I create a notebook from inside the script only and do all test cases on it, then delete it. This way no changes to the git history. What do you think?

Reply:
> A solution I was thinking of was, I create a notebook from inside the script only and do all test cases on it, then delete it. This way no changes to the git history. What do you think?

Yes, I think this could be a good way